File: 658/Solution.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        int start = 0;
        int end = arr.length-1;
        List<Integer> ar = new ArrayList<Integer>();
        
        while((end - start) > k-1) {
            if (Math.abs(arr[start] - x) <= Math.abs(arr[end] - x)) {
                end--;
            }
            else {
                start++;
            }
        }
        
        for (int i=0;i<k;i++) {
            ar.add(arr[i+start]);
        }
        
        return ar;
    }
}

File: 658/Solution.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        y = sorted([(a,abs(a-x)) for a in arr], key=lambda x: x[1])
        return sorted([a[0] for a in y][:k])
        